quixstreams/checkpointing/checkpoint.py

```diff
@@ -155,17 +162,56 @@ def commit(self):
         unproduced_msg_count = self._producer.flush()
         if unproduced_msg_count > 0:
             raise CheckpointProducerTimeout(
-                f"'{unproduced_msg_count}' messages failed to be produced before the producer flush timeout"
+                f"'{unproduced_msg_count}' messages failed to be produced before "
+                f"the producer flush timeout"
             )
 
-        # Get produced offsets after flushing the producer
-        produced_offsets = self._producer.offsets
+        logger.debug("Checkpoint: flushing sinks")
+        sinks = self._sink_manager.sinks
+        # Step 3. Flush sinks
+        for (topic, partition), offset in self._tp_offsets.items():
+            for sink in sinks:
+                if self._pausing_manager.is_paused(topic=topic, partition=partition):
+                    # The topic-partition is paused, skip flushing other sinks for
+                    # this TP.
+                    # Also drop the batch to reprocess it later.
+                    # Note: when flushing multiple sinks for the same TP, some
+                    # of them can be flushed before one of the sinks is backpressured.
+                    sink.on_paused(topic=topic, partition=partition)
+                    break
```

Comment:
Should this be `continue` ? I think we want to trigger the `on_paused` action of all sinks ?

Reply:
That's right, nice catch 👍 .

Reply:
Fixed and added a test for that